mock/py/mockbuild/config.py

```diff
@@ -629,7 +629,8 @@ def set_config_opts_per_cmdline(config_opts, options, args):
     if options.mode == 'yum-cmd':
         config_opts['package_manager'] = 'yum'
     if options.mode == 'dnf-cmd':
-        config_opts['package_manager'] = 'dnf'
+        # If config specifies a package manager (e.g. dnf5), leave it unchanged
+        config_opts.setdefault('package_manager', 'dnf')
```

Comment:
Or we can be more explicit if you want to, and do something like

```python
if config_opts['package_manager'] not in ['dnf', 'dnf5']:
    config_opts['package_manager'] = 'dnf'
```

Reply:
Huh, I thought these --yum-cmd/--dnf-cmd commands are aliases :-( this current implementation overwriting the package_manager config doesn't make much sense to me.

Anyway, what is changing now is that package_manager=yum doesn't get overwritten with `dnf` when --dnf-cmd is used.  What is your opinion on this?

Reply:
> Anyway, what is changing now is that package_manager=yum doesn't get overwritten with dnf when --dnf-cmd is used. What is your opinion on this?

This would IMHO be solved if we used the code that I am proposing in the comment above.

Reply:
I know but the code I am suggesting in the comment above:

```python
if config_opts['package_manager'] not in ['dnf', 'dnf5']:
    config_opts['package_manager'] = 'dnf'
```

will override `yum` with `dnf`. But it won't override `dnf5` with `dnf` ... that may be what we won't, or not?

Reply:
Updated, PTAL